### ISSUE TITLE
from torch_peaq import PEAQBasic

### ISSUE THREAD

**Issue by lixinghe1999** [woongzip1/UniverSR#7]
Dear authors and other community members,
here is a simple question, what's torch_peaq and how can i install it.
thanks for your help.

**Comment by deton24** [comment]
Just `pip install torch-pesq`
But it's a loss function for speech. So maybe if you don't train for
speech, don't bother.

czw., 14 maj 2026, 10:29 użytkownik Lixing He ***@***.***>
napisał:

**Comment by woongzip1** [comment]
Hi, just to clarify, `torch_peaq` is different from `torch-pesq`.

`torch_peaq` is an internal implementation we developed for evaluating the **2f-model**. While the 2f-model itself is based on the [SEBASS project](https://www.audiolabs-erlangen.de/resources/2019-WASPAA-SEBASS/) by AudioLabs Erlangen, we modified their code into our own PyTorch-based implementation for internal use. Please note that this specific `torch_peaq` module is not publicly released at this time.

In our experiments, we compared various perceptual metrics such as ViSQOL and PESQ (which is primarily for speech). We found that the 2f-model scores align most consistently with human perceptual quality for our task, which is why we chose it for our evaluation.

Best regards.

**Comment by oulianov** [comment]
Hey @woongzip1, thank you for the details. 
@lixinghe1999 raises a great point that this would be very helpful to reproduce your results, but also to help general research in audio quality. 
Thank you for the link: from what i see, there is a partial Matlab implementation linked. Making the 2f  evaluation function available in Python would be a great contribution to the open source community!

**Comment by woongzip1** [comment]
Hey @oulianov, thanks for bringing this up, and @lixinghe1999 for the original question.

Most 2f-model implementations rely on MATLAB, or people just end up writing their own. That's exactly why we implemented our own version in Python. Unfortunately the code was written by a colleague rather than by me, so I'm not in a position to open-source it at the moment.

As an alternative, I'd recommend seabass by @kwatcharasupat: https://github.com/kwatcharasupat/2f-model. It's a pure Python implementation validated against the reference values from AudioLabs.

One caveat worth knowing: AudioLabs notes that MOVs differ between PEAQ implementations, which shifts the 2f-model output, so scores may not match ours exactly.

Thanks again for the interest in our work. If our implementation becomes available to release, I'll follow up here.
